tests/integration/net/state/simple/replicator/with_create_update_test.go

```diff
@@ -141,17 +141,20 @@ func TestP2POneToOneReplicatorDoesNotUpdateDocExistingOnlyOnTarget(t *testing.T)
 					}
 				`,
 			},
-			testUtils.ConfigureReplicator{
-				SourceNodeID: 0,
-				TargetNodeID: 1,
-			},
 			testUtils.CreateDoc{
 				// This document is created in all nodes
 				Doc: `{
 					"Name": "John",
 					"Age": 21
 				}`,
 			},
+			testUtils.ConfigureReplicator{
```

Comment:
todo: This is changing what the test tests, and if there is flakiness here, it should probably be handled with a `WaitForSync` action instead.

That said, where have you seen this flakiness? Do you have an issue or an error log?

Reply:
The CLI tests fail with the original test setup. If the replicator syncs the document before the second node attempts to create it, then it will fail. I will swap it for a `WaitForSync` like you suggested.

Reply:
But only the CLI tests failed/were flaky?  That suggests a problem with the production code, not the test.  

If the other mutation types are not flaky, and the CLI is, the CLI is not behaving the same way as the others.

Reply:
I believe this is due to the CLI being the slowest implementation. If you look through the original test steps you can see where the race condition happens. The document syncs from node 0 to node 1 before node 1 has the chance to create it.

Reply:
> I believe this is due to the CLI being the slowest implementation to test.

That does not make sense to me though, as the slowness is outside of the execution bounds in which speed matters (RE race), and if anything the CLI should be more resilient to flakiness here, as there is a larger gap between submitting the Create, and the Update.

Where x = main test thread, y = create async stuff, z = update async stuff, * = mutation type specific x stuff:
```
x * // Create doc action start
x *
x y // 
x y
x y
x y
x y // Create doc action end
x * // Update doc action start
x *
x z
x z
x z
``` 
When running CLI tests * may occupy more time, but that should only have the same effect as adding in a wait/sleep, and should reduce the risk of hitting a race condition (where y and z overlap).

From memory I also believe that the `ConfigureReplicator` action includes a wait anyway.

`TestP2PPeerReplicatorWithUpdate` also seems to run fine with the same create-config-update flow?

Reply:
I created an issue to start tracking tests with potential race condition #1935